src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -266,14 +285,33 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     // - RoutedEventArgs: not used
     // Return Value:
     // - <none>
-    void TermControl::_CloseSearchBoxControl(const winrt::Windows::Foundation::IInspectable& /*sender*/, RoutedEventArgs const& /*args*/)
+    void TermControl::_CloseSearchBoxControl(const winrt::Windows::Foundation::IInspectable& /*sender*/,
+                                             RoutedEventArgs const& /*args*/)
     {
         _searchBox->Visibility(Visibility::Collapsed);
 
         // Set focus back to terminal control
         this->Focus(FocusState::Programmatic);
     }
 
+    // Method Description:
+    // - This event handler works to give the search box an attempt to process
+    //   keybindings. Most notably, we need this for the findPrevious and
+    //   findNext actions to be actionable from within the search box.
+    void TermControl::_SearchKeyHandler(Windows::Foundation::IInspectable const& /*sender*/,
+                                        Input::KeyRoutedEventArgs const& e)
+    {
+        auto modifiers = _GetPressedModifierKeys();

```

Comment:
```suggestion
        const auto modifiers = _GetPressedModifierKeys();
```

Reply:
@zadjii-msft - I believe that we need to agree on the approach, as in CmdPal I did something else: CmdPal doesn't get the bindings but gets the keymap and uses it in its internal PreviewKeyHandler: if the action is supported by the CmdPal (like switch tab) then it is handled, otherwise the event is handled externally.

If you prefer the approach used here, we should probably adapt the CmdPal as well.

Reply:
@Don-Vito So, I don't think you're wrong. The trick here is that the `TermControl` only knows about a  `...TerminalControl.IKeyBindings`, not a whole `KeyMapping`. I think yours is probably _better_, it's more specific. That being said, I don't want all consumers of a TerminalControl to need to implement a whole `KeyMapping` themselves. (Like I don't want force VS to conform to the implementation of keybindings that the Windows Terminal has chosen, if that makes sense).

TermControl's key interactions are likely to get blown up anyways with the work I'm planning for #5000, so I'm not too worried about this right now. Good chance it's gonna have to be re-written anyways 😅